crates/noirc_evaluator/src/ssa_refactor/ir/function_inserter.rs

```diff
@@ -13,6 +13,7 @@ use super::{
 /// The FunctionInserter can be used to help modify existing Functions
 /// and map old values to new values after re-inserting optimized versions
 /// of old instructions.
+/// TODO: optimized meaning calls to the simplify method et al?
```

Comment:
"optimized" here just meaning a new instruction that some optimization pass wants to map to an existing instruction. For example, loop unrolling will re-insert each instruction in the body of a loop N times and it must keep track of this mapping each time in case the old instruction id is referenced later in the same loop iteration.

The new instruction that is mapped to is generally expected to either be an equivalent instruction or a more optimized form but I suppose it isn't strictly necessary as long as both have the same number of results.